openpgp/ecdh/ecdh.go

```diff
@@ -145,7 +155,9 @@ func buildKey(pub *PublicKey, zb []byte, curveOID, fingerprint []byte, stripLead
 	}
 	kdf := pub.KDF.ReplacementKDFParams
 	if kdf == nil {
-		kdf = pub.KDF.write()
+		if err := pub.KDF.serialize(param); err != nil {
+			return nil, err
+		}
 	}
 	if _, err := param.Write(kdf); err != nil {
 		return nil, err
```

Comment:
If `ReplacementKDFParams` is nil, we're calling `Write(nil)` here - which I think works by accident when armoring, but idk if other implementations of `Write` also accept `nil`. So this should be something like:

```go
if pub.KDF.ReplacementKDFParams != nil {
	if _, err := param.Write(pub.KDF.ReplacementKDFParams); err != nil {
		return nil, err
	}
} else {
	if err := pub.KDF.serialize(param); err != nil {
		return nil, err
	}
}
```